### PR TITLE
allow environment variables in request body.

### PR DESCRIPTION
**Pull request by 0xc0Der**
solves issue #1834.

### CODE REVIEW

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
There are multiple things wrong with this PR.

First of all the functionality does not work.

Steps for reproduction:
recording

(open image in new tab for more clarity)

Secondly, this is breaking functionality with FormData editor when using environment variables.

Steps of reproduction:
recording

(open image in a new tab for more clarity)

**Comment by 0xc0Der** [comment]
hmmm. investigating.

**Comment by 0xc0Der** [comment]
fixed.
- the first problem was because u did not track env vars changes.
- the second is not mine. it in the `EnvInput` code.

**Comment by 0xc0Der** [comment]
i guess i missed up the branch. :smile:

File: packages/hoppscotch-app/components/http/RawBody.vue [hunk 1]
```diff
@@ -111,6 +119,25 @@ useCodemirror(
...
+}
+
+watch(envVariables, () => {
+  updateBody(rawBodyData.value)
+})
+
+watch(rawBodyData, (newValue) => {
+  updateBody(newValue)
+})
+
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/http/RawBody.vue:140]: This is totally not the right approach to implementing this in Hoppscotch.

Hoppscotch processes requests in a step-by-step process which are broadly classified into.

- Composing (The user is still building the request, did not press the Send button yet)
- Initial Validation (Deciding whether the request should even run, when the Send button is pressed)
- Pre-Request Script Execution (A Pre-Request script is executed if defined)
- Environment Injection (After pre-request script we have all the data necessary to fill the environment variable holes, we fill them)
- Execution (Actual network request is ran)

When the send request action is fired, the pre-request scripts are initially ran. [Pre-Request scripts](https://github.com/hoppscotch/hoppscotch/wiki/Pre-Request-Scripts) allow for environment variables to be created and updated dynamically, those environments are all aggregated into an ephemeral list of environmental variables as specified here https://github.com/hoppscotch/hoppscotch/blob/6d54f21c1eed4df152b95e7c79a92d9b0fe1ae09/packages/hoppscotch-app/helpers/RequestRunner.ts#L49-L53

Your code tries to resolve environment variables during the composing phase itself, which doesn't take into account the dynamic variables which can be defined by the pre-request scripts.

**Review by AndrewBastin — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-app/components/http/RawBody.vue (hunk 1 above)
Comment (0xc0Der) [reply packages/hoppscotch-app/components/http/RawBody.vue:140]: noted. re-investigating.

**Review by AndrewBastin — APPROVED** [review]
LGTM 💯 

Once again thank you for the PR @0xc0Der 🤗 🤗

**Pull request merged**